assembly/src/parsers/io_ops.rs

```diff
@@ -256,69 +276,144 @@ pub fn parse_loc_storew(op: &Token) -> Result<Node, ParsingError> {
 // HELPER FUNCTIONS
 // ================================================================================================
 
-fn parse_constants(op: &Token) -> Result<Vec<Felt>, ParsingError> {
-    let mut constants = Vec::new();
-    let param_idx = 1;
-    let param_count = op.num_parts() - param_idx;
+/// Parses a list of parameters (each of which could be in decimal or hexadecimal form) and returns
+/// an appropriate push instruction node.
+fn parse_param_list(op: &Token) -> Result<Node, ParsingError> {
+    let mut values = Vec::with_capacity(op.num_parts() - 1);
+    for (param_idx, &param_str) in op.parts().iter().enumerate().skip(1) {
+        let value = match param_str.strip_prefix("0x") {
+            Some(param_str) => parse_hex_value(op, param_str, param_idx),
+            None => parse_checked_param(op, param_idx, 0, Felt::MODULUS - 1),
+        };
+        values.push(value);
+    }
```

Comment:
I think here as well we could build this as an iterator and then pass this iterator to `build_push_many_instruction()` instead of first building a vector and then converting it to an iterator.